### PR TITLE
Fixes several sphinx warnings.

### PR DESCRIPTION
**Pull request by melissawm**
# Description
This fixes several straightforward Sphinx warnings and simplifies the docs build. 

Before: 384 warnings
Now: 290 warnings

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Locally, you can do `make docs` to see the build with these fixes. In addition, if you want to confirm that this works when deployed to napari.github.io, you can:
1. Clone both [napari/napari](https://github.com/napari/napari) and [napari/napari.github.io](https://github.com/napari/napari.github.io)
2. Go into the napari/napari folder and run
   ```bash
   $ pip install -e '.[all]'
   $ make docs
   ```
3. View changes rendered in "_build/html"
4. Run `python docs/copy-docs.py` (this will copy the appropriate sources into the napari/napari.github.io folder and update the `napari/napari.github.io/_toc.yml` file
5. Go into the napari/napari.github.io folder and run
	```bash
	$ yarn dev
	```
6. Confirm that this command renders the appropriate outputs.

### A few things to note
- Check out the difference between https://napari.org/api/stable/napari.Viewer.html and this build (http://localhost:8080/api/stable/napari.Viewer.html):

**Old**
napariold

**New**
naparinew
naparinew2

- Currently, there is a page that appears twice in the ToC (`api/stable/napari.html`). This is due to sphinx not being able to process a `:hidden:` option for the autodoc directive and I'm looking for solutions.
- The remaining warnings are mostly of two types: Sphinx not being able to parse typehints, and repeated references to autogenerated docs. Both should be fixed in a later PR.
 
## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: docs/_scripts/update_preference_docs.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 GUIDES_PATH = DOCS / "guides"
 IMAGES_PATH = DOCS / "images" / "_autogenerated"
 IMAGES_PATH.mkdir(parents=True, exist_ok=True)
-PREFERENCES_TEMPLATE = """(preferences)=
+PREFERENCES_TEMPLATE = """(napari-preferences)=
```
Comment (melissawm) [inline docs/_scripts/update_preference_docs.py:17]: Fixes the problem mentioned [here](https://github.com/napari/napari/issues/3654#issuecomment-973317892)

File: docs/_templates/autosummary/class.rst [hunk 2]
```diff
@@ -27,9 +26,10 @@
    .. rubric:: {{ _('Attributes') }}
 
    .. autosummary::
-      :inherited-members:
```
Comment (melissawm) [inline docs/_templates/autosummary/class.rst:30]: This was not working anyway, but can anyone please confirm if the level of documented objects is sufficient without this option?

File: docs/api/index.rst [hunk 3]
```diff
@@ -37,8 +37,8 @@ For those wishing to add custom functionality to their project.
 
 
 .. autosummary::
-   :hidden:
```
Comment (melissawm) [inline docs/api/index.rst:40]: This worked in part, but was generating warnings. I think there is a better solution but I'm still looking for it.

File: docs/_scripts/update_preference_docs.py (hunk 1 above)
Comment (tlambert03) [reply docs/_scripts/update_preference_docs.py:17]: just curious, did you ever verify whether this was still a problem if the preferences page actually gets built?  (I would assume that intersphinx would look for internal links before looking externally?

**Review by tlambert03 — APPROVED** [review]

**Review by goanpeca — APPROVED** [review]
Awesome work @melissawm 🚀

**Comment by sofroniewn** [comment]
I will get this in now!

**Pull request merged**